doc/ECMP/Class-Based-Forwarding.md

```diff
@@ -0,0 +1,177 @@
+# Class-based Forwarding
+
+Title       | Class-based Forwarding
+------------|----------------
+Authors     | Cisco
+Status      | In review
+Type        | Standards track
+Created     | 04/14/2021
+SAI-Version | 1.8
+
+Class-based forwarding provides a method to steer traffic among multiple paths through the network by policy rather than, or in combination with, traditional ECMP/UCMP flow-hashing.
+
+A new type of next-hop group is proposed:
+
+```
+typedef enum _sai_next_hop_group_type_t
+{
+...
+    /** Next hop group is class-based, with members selected by Forwarding class */
+    SAI_NEXT_HOP_GROUP_TYPE_CLASS_BASED,
+...
+} sai_next_hop_group_type_t;
+```
+
+The behavior of SAI_NEXT_HOP_GROUP_TYPE_CLASS_BASED differs from the traditional SAI_NEXT_HOP_GROUP_TYPE_ECMP, in that each packet will have a Forwarding class that directly chooses next-hop group member index.
+
+This is accomplished by mapping each forwarding-class to the group member index.
+
+```
+    /**
+     * @brief Forwarding-class to index map
+     *
+     * @type sai_map_list_t
+     * @flags CREATE_AND_SET
+     * @default empty
+     * @validonly SAI_NEXT_HOP_GROUP_ATTR_TYPE == SAI_NEXT_HOP_GROUP_TYPE_CLASS_BASED
+     */
+    SAI_NEXT_HOP_GROUP_ATTR_FORWARDING_CLASS_TO_INDEX_MAP,
+```
```

Comment:
From OCP call: Use separate map object to enable CRM

Reply:
Also, it would be a good idea to define capability attributes for the number of forwarding classes and number of indices supported by the SAI implementation.

Reply:
Added documentation of switch attribute for max forwarding classes.

Reply:
Implemented separate map object.